trlx/utils/modeling.py

```diff
@@ -34,3 +55,49 @@ def flatten_dict(
         else:
             items.append((new_key, v))
     return dict(items)
+
+
+def log_stat(stats: dict, name: str, xs: torch.Tensor, mask: torch.Tensor, n: int):
+    mean = (xs * mask).sum() / n
+    stats.update(
+        {
+            f"{name}/mean": mean,
+            f"{name}/min": torch.where(mask.bool(), xs, np.inf).min(),
+            f"{name}/max": torch.where(mask.bool(), xs, -np.inf).max(),
+            f"{name}/std": torch.sqrt(((xs - mean) * mask).pow(2).sum() / n),
+        }
+    )
+
+
+class RunningMoments:
+    def __init__(self):
+        """
+        Calculates the running mean and standard deviation of a data stream. Modified version of
+        https://github.com/DLR-RM/stable-baselines3/blob/a6f5049a99a4c21a6f0bcce458ca3306cef310e0/stable_baselines3/common/running_mean_std.py
+        """
+        self.mean = 0
+        self.std = 1
+        self.var = 1
+        self.count = 1e-24
+
+    def update(self, xs: torch.Tensor) -> Tuple[float, float]:
+        """Updates running moments from batch's moments computed across ranks"""
+        if dist.is_initialized():
+            xs_mean, xs_var, xs_count = get_global_statistics(xs)
+        else:
+            xs_count = xs.numel()
+            xs_var, xs_mean = torch.var_mean(xs, unbiased=False)
+
+        delta = xs_mean - self.mean
+        tot_count = self.count + xs_count
+
+        m_a = self.var * self.count
```

Comment:
I hate having lots of math and no intuitive explanation of what the math is doing as comments. Please fix.